extensions/src/test/java/org/pentaho/platform/plugin/services/metadata/PentahoMetadataDomainRepositoryTest.java

```diff
@@ -14,121 +14,1092 @@
+      put( PROPERTY_NAME_DOMAIN_ID, "someId" );
+      put( PROPERTY_NAME_TYPE, TYPE_DOMAIN );
+    } };
+
+    IUnifiedRepository repository = Mockito.mock( IUnifiedRepository.class );
+    when( repository.getFileMetadata( any() ) ).thenReturn( fileMetadataLegacy );
+
+    XmiParser xmiParser = Mockito.mock( XmiParser.class );
+
+    PentahoMetadataDomainRepository pmdr = new PentahoMetadataDomainRepository( repository, null, xmiParser, null, systemConfig );
+
+    // EXECUTE
+    Map<String, Serializable> fileMetadataActual = pmdr.getFileMetadataHelper( serializableMetadataDir );
+
+    // VERIFY
+    assertTrue( equalMaps( fileMetadataExpected, fileMetadataActual ) );
+  }
+
+  Domain createDSWTestObject() {
+    Domain domain = new Domain();
+
+    List<LogicalModel> models = new ArrayList<>();
+
+    LogicalModel lm1 = new LogicalModel();
+    lm1.setProperty( "AGILE_BI_GENERATED_SCHEMA", "TRUE" );
+    models.add( lm1 );
+
+    LogicalModel lm2 = new LogicalModel();
+    lm2.setProperty( "WIZARD_GENERATED_SCHEMA", "TRUE" );
+    models.add( lm2 );
+
+    domain.setLogicalModels( models );
+
+    return domain;
+  }
+
+  Domain createMetadataTestObject() {
+    Domain domain = createDSWTestObject();
+    domain.getLogicalModels().clear();
+
+    return domain;
+  }
+
+  public ISystemConfig createSystemConfigTestObject( HashMap<String, Serializable> map ) throws Exception {
+    Properties properties = new Properties();
+    properties.putAll( map );
+    return createSystemConfigTestObject( properties );
+  }
+
+  public ISystemConfig createSystemConfigTestObject( Properties properties ) throws Exception {
+    IConfiguration configuration = Mockito.mock( IConfiguration.class );
+    when( configuration.getId() ).thenReturn( "system" );
+    Mockito.when( configuration.getProperties() ).thenReturn( properties );
+
+    List<IConfiguration> configs = new ArrayList<>();
+    configs.add( configuration );
+    return new SystemConfig( configs );
+  }
+
+  /**
+   * Tests if two maps are equal in size and values.
+   * @param m1
+   * @param m2
+   * @param <K>
+   * @param <V>
+   * @return true is equal in size and values, false otherwise.
+   */
+  public static <K, V> boolean equalMaps( Map<K, V> m1, Map<K, V> m2 ) {
+    if ( m1 == null && m2 == null ) return true;
+    if ( m1 == null || m2 == null ) return false;
+    if ( m1.size() != m2.size() ) return false;
+
+    for ( K key: m1.keySet() )
+      if ( !m1.get( key ).equals( m2.get( key ) ) )
+        return false;
+    return true;
+  }
+
+  /**
+   * Tests if two maps are equal in size and values.
+   */
+  public static class MapMatcher<K, V> implements ArgumentMatcher<Map<K, V>> {
+
+    private Map<K, V> leftMap;
+
+    public MapMatcher( Map<K, V> map ) {
+      this.leftMap = map;
+    }
+
+    @Override
+    public boolean matches( Map<K, V> rightMap ) {
+      return equalMaps( leftMap, rightMap );
+    }
+  }
```

Comment:
Is this class being used?

Reply:
No `MapMatcher` is not being used. I was going to use in an a lamda expression and/or stream.
will change.

Reply:
see commit: https://github.com/pentaho/pentaho-platform/pull/5124/commits/89886462daa3c1d9db9fa368bdcd3fa7ceb27804